src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -509,40 +521,43 @@
-         {:keys [code details] :as error-response} (-> data :ErrorResponse)
-         skip-processing-suggested-routes?         (get-in db
-                                                           [:wallet :ui :send
-                                                            :skip-processing-suggested-routes?])
-         process-response?                         (and (not swap?)
-                                                        (not skip-processing-suggested-routes?))]
-     (if (and (not swap?) error-response)
-       (let [error-message (if (= code "0") "An error occurred" details)]
-         (log/error "failed to get suggested routes (async)"
-                    {:event :wallet/handle-suggested-routes
-                     :error error-message})
-         {:fx [(cond
-                 swap?
-                 [:dispatch [:wallet/swap-proposal-error error-message]]
-                 process-response?
-                 [:dispatch [:wallet/suggested-routes-error error-message]])]})
-       (let [best-routes-fix (comp ->old-route-paths
-                                   remove-invalid-bonder-fees-routes
-                                   remove-multichain-routes)
-             candidates-fix  (comp ->old-route-paths
-                                   remove-invalid-bonder-fees-routes)
-             routes          (-> data
-                                 (data-store/rpc->suggested-routes)
-                                 (update :best best-routes-fix)
-                                 (update :candidates candidates-fix))]
-         {:fx [(cond
-                 swap?
-                 [:dispatch [:wallet/swap-proposal-success routes]]
-                 process-response?
-                 [:dispatch [:wallet/suggested-routes-success routes]])]})))))
+   (let [{send :send swap? :swap} (-> db :wallet :ui)
+         skip-processing-routes?  (:skip-processing-suggested-routes? send)]
+     (when (or swap? (not skip-processing-routes?))
+       (let [{error-code :code
+              :as        error} (:ErrorResponse data)
+             enough-assets?     (not (and (:Best data) (= error-code "WR-002")))
+             failure?           (and error enough-assets? (not swap?))
+             error-message      (if (zero? error-code) "An error occurred" (:details error))]
+         (when failure?
+           (log/error "failed to get suggested routes (async)"
+                      {:event :wallet/handle-suggested-routes
+                       :error error-message}))
+         {:fx [[:dispatch
+                (cond
+                  (and failure? swap?) [:wallet/swap-proposal-error error-message]
+                  failure?             [:wallet/suggested-routes-error error-message]
+                  swap?                [:wallet/swap-proposal-success (fix-routes data)]
+                  :else                [:wallet/suggested-routes-success (fix-routes data)
+                                        enough-assets?])]]})))))
```

Comment:
Modified this event to make the logic a bit clearer, hoping now it's easier to understand since we don't have many conditions to check.

It's been also fixed to properly support the error code `WR-002` returned by the status-go when the user doesn't have enough assets